src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -766,14 +759,27 @@ public static FormulaValue Mid(IRContext irContext, FormulaValue[] args)
             }
 
             var source = (StringValue)args[0];
-            var start0Based = (int)(start.Value - 1);
-            if (source.Value == string.Empty || start0Based >= source.Value.Length)
+            double startValue = start.Value;
+            if (startValue > int.MaxValue)
```

Comment:
>startValue [](http://example.com/codeflow?start=16&length=10)

Can we use the new helper  here?
And that would avoid the (int) cast on 768

Reply:
update here